### ISSUE TITLE
[Crash]: Linux Client crashes on error https://foo.server/error?type=channel_not_found&returnTo=/barGroup/channels/town-square

### ISSUE THREAD

**Issue by Scoeerg** [mattermost/desktop#3139]
### Before you file a crash report

- [X] I have checked the [issue tracker](https://github.com/mattermost/desktop/issues) and have not found an issue that matches the one I'm filing.
- [X] This issue doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/mattermost-server/issues).
- [X] I have read the [contribution guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md).

### Mattermost Desktop Version

5.9.0

### Operating System

Ubuntu Linux 22.04 LTS x64

### What type of crash did you experience?

System crash (The application quit unexpectedly with no warning, or the operating system reported a crash)

### Crash report details

**tl;dr:** 

Linux Client crashes on error https://foo.server/error?type=channel_not_found&returnTo=/barGroup/channels/town-square

**Details:**

Given a direct chat between two users `A` and `B`. If there exists a link to channel in the direct chat between those two users

hey, look into ~myChannel

it of course links to https://foo.server/barGroup/channels/mychannel. Click on it, and the channel `mychannel` opens. If the user A is also in another group `foobar_group`

https://foo.server/foobar_group/

where the channel `mychannel` does not exist and he clicks on the same link within the direct chat, mattermost throws the exception

https://foo.server/error?type=channel_not_found&returnTo=/foobar_group/channels/town-square

which is fine in browser, but it crashes the Linux client.

### Log Output

My logs this morning. Crash happened around 11.30 a.m. Maybe logged as `[2024-09-03 11:31:27.322] [error] [WebContentsEventM...] [RPTU] [TAB_MESSAGING] [renderer] Uncaught (in promise) Error: Invalid or missing user_id parameter in request URL.`?

```shell
[2024-09-03 08:44:24.713] [error] Logger Log level set to: info
[2024-09-03 08:44:24.830] [warn]  [App.Initialize] Current working directory is /home/eric, changing into /snap/mattermost-desktop/762/opt/Mattermost
[2024-09-03 08:44:25.166] [info]  [App.Config] config.autostart has been configured: true
[2024-09-03 08:44:25.233] [info]  [App.Initialize] Autoupgrade disabled: false
[2024-09-03 08:44:25.288] [info]  [ServerDropdownView] init
[2024-09-03 08:44:25.329] [warn]  [i18nManager] Failed to set new language en-US
[2024-09-03 08:44:25.330] [warn]  [i18nManager] Failed to set new language US
[2024-09-03 08:44:25.485] [info]  [App.Config] config.autostart has been configured: true
[2024-09-03 08:44:25.836] [info]  [App.Config] config.autostart has been configured: true
[2024-09-03 08:44:26.630] [error] [WebContentsEventM...] [RPTU] [TAB_MESSAGING] [renderer] Uncaught (in promise) Error: Invalid or missing user_id parameter in request URL.
[2024-09-03 08:59:57.047] [error] [WebContentsEventM...] [RPTU] [TAB_MESSAGING] [renderer] Uncaught (in promise) Error: Invalid or missing user_id parameter in request URL.
[2024-09-03 09:00:03.103] [error] [WebContentsEventM...] [RPTU] [TAB_MESSAGING] [renderer] Uncaught (in promise) Error: Invalid or missing user_id parameter in request URL.
[2024-09-03 10:02:51.906] [error] [WebContentsEventM...] [RPTU] [TAB_MESSAGING] [renderer] Uncaught (in promise) Error: Invalid or missing user_id parameter in request URL.
[2024-09-03 11:03:19.667] [error] [WebContentsEventM...] [RPTU] [TAB_MESSAGING] [renderer] Uncaught (in promise) Error: Invalid or missing user_id parameter in request URL.
[2024-09-03 11:05:29.112] [error] [WebContentsEventM...] [RPTU] [TAB_MESSAGING] [renderer] Uncaught (in promise) Error: Invalid or missing user_id parameter in request URL.
[2024-09-03 11:09:16.257] [error] [WebContentsEventM...] [RPTU] [TAB_MESSAGING] [renderer] Uncaught (in promise) Error: Invalid or missing user_id parameter in request URL.
[2024-09-03 11:10:23.646] [error] [WebContentsEventM...] [RPTU] [TAB_MESSAGING] [renderer] Uncaught (in promise) Error: Invalid or missing user_id parameter in request URL.
[2024-09-03 11:10:30.219] [error] [WebContentsEventM...] [RPTU] [TAB_MESSAGING] [renderer] Uncaught (in promise) Error: Invalid or missing user_id parameter in request URL.
[2024-09-03 11:10:33.322] [error] [WebContentsEventM...] [RPTU] [TAB_MESSAGING] [renderer] Uncaught (in promise) Error: Invalid or missing user_id parameter in request URL.
[2024-09-03 11:10:40.450] [error] [WebContentsEventM...] [RPTU] [TAB_MESSAGING] [renderer] Uncaught (in promise) Error: Invalid or missing user_id parameter in request URL.
[2024-09-03 11:10:44.058] [error] [WebContentsEventM...] [RPTU] [TAB_MESSAGING] [renderer] Uncaught (in promise) Error: Invalid or missing user_id parameter in request URL.
[2024-09-03 11:13:07.639] [error] [WebContentsEventM...] [RPTU] [TAB_MESSAGING] [renderer] Uncaught (in promise) Error: Invalid or missing user_id parameter in request URL.
[2024-09-03 11:19:01.582] [error] [WebContentsEventM...] [RPTU] [TAB_MESSAGING] [renderer] Uncaught (in promise) Error: Invalid or missing user_id parameter in request URL.
[2024-09-03 11:31:25.744] [error] Logger Log level set to: info
[2024-09-03 11:31:25.827] [warn]  [App.Initialize] Current working directory is /home/eric, changing into /snap/mattermost-desktop/762/opt/Mattermost
[2024-09-03 11:31:25.953] [info]  [App.Config] config.autostart has been configured: true
[2024-09-03 11:31:25.971] [info]  [App.Initialize] Autoupgrade disabled: false
[2024-09-03 11:31:26.006] [info]  [ServerDropdownView] init
[2024-09-03 11:31:26.052] [warn]  [i18nManager] Failed to set new language en-US
[2024-09-03 11:31:26.052] [warn]  [i18nManager] Failed to set new language US
[2024-09-03 11:31:26.199] [info]  [App.Config] config.autostart has been configured: true
[2024-09-03 11:31:26.540] [info]  [App.Config] config.autostart has been configured: true
[2024-09-03 11:31:27.322] [error] [WebContentsEventM...] [RPTU] [TAB_MESSAGING] [renderer] Uncaught (in promise) Error: Invalid or missing user_id parameter in request URL.
[2024-09-03 11:31:32.772] [error] [WebContentsEventM...] [RPTU] [TAB_MESSAGING] [renderer] Uncaught (in promise) Error: Invalid or missing user_id parameter in request URL.
```

### Additional Information

hope it helps :)

**Comment by devinbinnie** [comment]
@Scoeerg Is this consistently reproducible? If so, can you try running the client from the command line and reproducing? There might be some additional logs there - it's looking like Chromium is crashing somehow.

**Comment by Scoeerg** [comment]
Sure, let me look into it tomorrow :)

**Comment by devinbinnie** [comment]
@Scoeerg Were you able to reproduce with additional information?

**Comment by Scoeerg** [comment]
Sorry, I could not do it before my vacation :/. I will be back September 30th.

**Comment by devinbinnie** [comment]
Closing as inactive. Feel free to message if this is still relevant.
